posts/2017-07-24-anatomy-of-a-serverless-app.md

```diff
@@ -0,0 +1,669 @@
+
+I had to make some choices, before I started development:
+
+* **Programming language**: [NodeJS](https://nodejs.org/en/) (my familiarity, all serverless platforms support it)
+* **Platform**: [AWS Lambda](https://aws.amazon.com/lambda/) (most popular and mature, lots of supporting services)
+* **Toolset**: [Serverless Framework](https://serverless.com/framework/) (opensource, repo with 17K+ stars, actively maintained, 2 week release cadence)
+
+## Setup
+
+The setup required to start development was straighforward:
+
+1. Install NodeJS: [download](https://nodejs.org/en/download/) or [using package manager](https://nodejs.org/en/download/package-manager/#osx)
+2. Install the [Serverless Framework](https://serverless.com/framework/): `npm install -g serverless`
+3. Setup an AWS account
+    * [Sign up for AWS](http://docs.aws.amazon.com/lambda/latest/dg/setting-up.html#setting-up-signup)
+    * [Create an IAM User](http://docs.aws.amazon.com/lambda/latest/dg/setting-up.html#setting-up-iam)
+    * [Install and setup AWS CLI](http://docs.aws.amazon.com/lambda/latest/dg/setup-awscli.html)
+
+The AWS setup is necessary so that we can deploy our serverless application on AWS Lambda. After creating the AWS IAM user, you will have to [configure the credentials](https://serverless.com/framework/docs/providers/aws/guide/credentials/) to give access to the Serverless Framework, for creating and managing resources on your behalf.
+
+You can use either of the options to configure the credentials:
+
+`aws configure` or `serverless config credentials`
+
+> Setting up an account with AWS and configuring was the most time-consuming and painful process, but as you will see, it gets better after that. So keep chugging along.
+
+## Creating the Project
+
+A little bit of planning on the project structure makes a lot of difference in visualizing the different parts of the system. We will build an application named `postman`, with a simple frontend using `curl` and an backend serverless email service, to send out emails to users. 
+
+Here is an example that works for me:
+
+```
+|-- postman
+  |-- README.md
+  |-- frontend
+  `-- services
+```
+where,
+
+  * `frontend` folder holds the frontend application
+  * `services` folder holds the serverless service(s)
+
+A structure like this provides clear separation for the non-serverless and serverless code for the overall application. The way the non-serverless portion of the application is written is totally your choice. In this post, we will primarily focus on the serverless portion of the application inside the `services` folder.
```

Comment:
The `frontend` part is confusing to me since it's not referenced in the post at all.